File: client/src/app/_models/hr/intervwItemCandidate.ts
import { IIntervwCandAttachment, IntervwCandAttachment } from "./intervwCandAttachment";
import { UserAttachment } from "./userAttachment";

export interface IIntervwItemCandidate {
     id: number;
     intervwItemId: number;
     interviewItemId: number;
     candidateId: number;
     personId: string;
     prospectiveCandidateId: number;
     applicationNo: number;
     candidateName: string;
     passportNo: string;
     scheduledFrom: Date;
     reportedAt?: Date;
     interviewedAt?: Date;
     interviewStatus: string;
     interviewerRemarks: string;
     attachmentFileNameWithPath: string;
}

export class IntervwItemCandidate implements IIntervwItemCandidate {
     id= 0;
     interviewItemId= 0;
     intervwItemId: number=0;
     candidateId= 0;
     prospectiveCandidateId = 0;
     personId = '';
     applicationNo= 0;
     candidateName= '';
     passportNo= '';
     scheduledFrom= new Date;
     reportedAt?=new Date;
     interviewedAt?=new Date;
     interviewStatus = '';
     interviewerRemarks= '';
     attachmentFileNameWithPath = '';
}

File: api/Entities/Admin/IntervwItemCandidate.cs
using api.Entities.HR;

namespace api.Entities.Admin
{
    public class IntervwItemCandidate: BaseEntity
    {
        public int InterviewItemId { get; set; }
        public DateTime ScheduledFrom { get; set; }
        //public DateTime ScheduledUpto { get; set; }
        public DateTime? ReportedAt { get; set; }  
        public DateTime? InterviewedAt { get; set;}
        public int CandidateId { get; set; }
        public string PersonId {get; set;}
        public int ProspectiveCandidateId { get; set; }
        public int ApplicationNo { get; set; }
        public string CandidateName { get; set; }
        public string InterviewerRemarks { get; set; }
        public string InterviewStatus { get; set; }
        public string AttachmentFileNameWithPath { get; set; }
    }
}